packages/eslint-plugin/docs/rules/restrict-plus-operands.md

```diff
@@ -18,70 +18,159 @@ This rule reports when a `+` operation combines two values of different types, o
 ### ❌ Incorrect
 
 ```ts
-var foo = '5.5' + 5;
-var foo = 1n + 1;
+let foo = '5.5' + 5;
+let foo = 1n + 1;
 ```
 
 ### ✅ Correct
 
 ```ts
-var foo = parseInt('5.5', 10) + 10;
-var foo = 1n + 1n;
+let foo = parseInt('5.5', 10) + 10;
+let foo = 1n + 1n;
 ```
 
 ## Options
 
-### `checkCompoundAssignments`
+:::caution
+We generally recommend against using these options, as they limit which varieties of incorrect `+` usage can be checked.
 
-Examples of code for this rule with `{ checkCompoundAssignments: true }`:
+For example, instead of enabling the `allow*` options to concatenate strings and other values, use APIs akin to `console.log` that allow passing multiple values:
```

Comment:
an alternative, more fleshed-out version with examples for consideration

```md
:::caution
We generally recommend against using these options, as they limit which varieties of incorrect `+` usage can be checked. This in turn severely limits the validation that the rule can do to ensure that resulting strings and numbers are correct.

Safer alternatives to using the `allow*` options include:
- Using `.toFixed()` to coerce numbers to well-formed string representations:
    ```ts
    const number = 1.123456789;
    const result = 'The number is ' + number.toFixed(2);
    // result === 'The number is 1.12'
    ```
- Calling `.toString()` on other types to mark explicit and intentional string coercion:
    ```ts
    const arg = '11';
    const regex = /[0-9]/;
    const result =
        'The result of ' + regex.toString() + '.test("' + arg + '") is ' + regex.test(arg).toString();
    // result === 'The result of /[0-9]/.test("11") is true'
    ```
- Using variadic forms of logging APIs to avoid needing to `+` values.
    ```ts
    // Remove this line
    console.log('The result is ' + true);
    // Add this line
    console.log('The result is', true);
    ```
```

<details>
<summary>Rendered</summary>

:::caution
We generally recommend against using these options, as they limit which varieties of incorrect `+` usage can be checked. This in turn severely limits the validation that the rule can do to ensure that resulting strings and numbers are correct.

Safer alternatives to using the `allow*` options include:
- Using `.toFixed()` to coerce numbers to well-formed string representations:
    ```ts
    const number = 1.123456789;
    const result = 'The number is ' + number.toFixed(2);
    // result === 'The number is 1.12'
    ```
- Calling `.toString()` on other types to mark explicit and intentional string coercion:
    ```ts
    const arg = '11';
    const regex = /[0-9]/;
    const result = 'The result of ' + regex.toString() + '.test("' + arg + '") is ' + regex.test(arg).toString();
    // result === 'The result of /[0-9]/.test("11") is true'
    ```
- Using variadic forms of logging APIs to avoid needing to `+` values.
    ```ts
    // Remove this line
    console.log('The result is ' + true);
    // Add this line
    console.log('The result is', true);
    ```

</details>

Reply:
Love it! My only change suggestion is moving the `console.log` one to be first because it's the most common easy workaround I see in the wild.